draft-ietf-quic-recovery.md

```diff
@@ -166,12 +157,50 @@ measure and report the delay from when a packet was received by the OS kernel,
 which is useful in receivers which may incur delays such as context-switch
 latency before a userspace QUIC receiver processes a received packet.
 
+
 # Loss Detection
 
-We now describe QUIC's loss detection as functions that should be called on
-packet transmission, when a packet is acked, and timer expiration events.
+## Overview {#overview}
+
+QUIC uses a combination of ack information and alarms to detect lost packets.
+An unacknowledged QUIC packet is marked as lost in one of the following ways:
+
+  * A packet is marked as lost if at least one packet that was sent a threshold
+    number of packets (kReorderingThreshold) after it has been
+    acknowledged. This indicates that the unacknowledged packet is either lost
+    or reordered beyond the specified threshold. This mechanism combines both
+    TCP's FastRetransmit and FACK mechanisms.
+
+  * If a packet is near the tail, where fewer than kReorderingThreshold packets
+    are sent after it, the sender cannot expect to detect loss based on the
+    previous mechanism. In this case, a sender uses both ack information and an
+    alarm to detect loss. Specifically, when the last sent packet is
+    acknowledged, the sender waits a short period of time to allow for
+    reordering and then marks any unacknowledged packets as lost. This mechanism
+    is based on the Linux implementation of TCP Early Retransmit.
+
+  * If a packet is sent at the tail, there are no packets sent after it, and the
+    sender cannot use ack information to detect its loss. The sender therefore
+    relies on an alarm to detect such tail losses. This mechanism is based on
+    TCP's Tail Loss Probe.
 
-## Constants of interest
+  * If all else fails, a Retransmission Timeout (RTO) alarm is always set when
```

Comment:
That's not really true.  We only arm the RTO if we've exceeded max TLP and early retransmit timer and time loss detection don't apply.